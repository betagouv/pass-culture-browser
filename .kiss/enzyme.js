import React from 'react'
import { shallow } from 'enzyme'

import MyComponent from '../MyComponent'

describe('src | components | MyComponent', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      const props = { errors: {} }
      const wrapper = shallow(<MyComponent {...props} />)
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
