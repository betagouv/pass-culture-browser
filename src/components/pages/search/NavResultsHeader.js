import React from 'react'
import PropTypes from 'prop-types'

// import get from 'lodash.get'

import { ROOT_PATH } from '../../../utils/config'

const NavResultsHeader = ({ category, description }) => {
  const searchTypeWithoutSpecialChar = category.replace(/Ã‰/g, 'E')
  const src = `${ROOT_PATH}/icons/img-${searchTypeWithoutSpecialChar}.png`
  const imgDescription = `Liste des offres de type ${category}`
  return (
    <div id="nav-results-header">
      <div
        className="flex-rows text-left is-absolute is-white-text ml24 mr48 mt20 mb32 "
        id="category-description"
      >
        <div className="flex-1 fs22">
          <span className="is-bold">
            {category}
          </span>
        </div>
        <div className="flex-1">
          {description}
        </div>
      </div>
      <img src={src} alt={imgDescription} />
    </div>
  )
}

NavResultsHeader.defaultProps = {
  category: null,
  description: null,
}

NavResultsHeader.propTypes = {
  category: PropTypes.string,
  description: PropTypes.string,
}

export default NavResultsHeader
