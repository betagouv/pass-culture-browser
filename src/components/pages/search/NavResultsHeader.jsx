import PropTypes from 'prop-types'
import React from 'react'

import { ROOT_PATH } from '../../../utils/config'

const NavResultsHeader = ({ category, description }) => {
  const src = `${ROOT_PATH}/icons/img-${category}-L.jpg`
  const imgDescription = `Liste des offres de type ${category}`

  return (
    <div
      id="nav-results-header"
      style={{
        backgroundImage: `url(${src})`,
      }}
      title={imgDescription}
    >
      <div id="category-description">
        <h2 className="nav-result-title">{category}</h2>
        <span className="nav-result-description">{description}</span>
      </div>
    </div>
  )
}

NavResultsHeader.defaultProps = {
  category: null,
  description: null,
}

NavResultsHeader.propTypes = {
  category: PropTypes.string,
  description: PropTypes.string,
}

export default NavResultsHeader
