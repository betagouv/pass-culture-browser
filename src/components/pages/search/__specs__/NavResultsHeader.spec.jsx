import { shallow } from 'enzyme'
import React from 'react'

import NavResultsHeader from '../NavResultsHeader'

describe('src | components | pages | search | NavResultsHeader', () => {
  let props

  beforeEach(() => {
    props = {
      category: 'Applaudir',
      description:
        'Voulez-vous suivre un géant de 12 mètres dans la ville ? Rire devant un seul-en-scène ? Rêver le temps d’un opéra ou d’un spectacle de danse, assister à une pièce de théâtre, ou vous laisser conter une histoire ?',
    }
  })

  it('should match the snapshot', () => {
    // when
    const wrapper = shallow(<NavResultsHeader {...props} />)

    // then
    expect(wrapper).toMatchSnapshot()
  })

  describe('render()', () => {
    it('should display the background image with the right description by default', () => {
      // given
      props.category = 'Écouter'
      props.description = 'Lorem Ipsum'
      const wrapper = shallow(<NavResultsHeader {...props} />)

      // when
      const img = wrapper.find('.nav-results-header').props()
      const title = wrapper.find('h2')

      // then
      expect(img.className).toBe('nav-results-header nav-results-Écouter')
      expect(title.text()).toBe('Écouter')
    })
  })
})
