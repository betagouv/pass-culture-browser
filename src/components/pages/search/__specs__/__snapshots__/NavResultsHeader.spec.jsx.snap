// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | search | NavResultsHeader should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavResultsHeader
    category="Applaudir"
    description="Voulez-vous suivre un géant de 12 mètres dans la ville ? Rire devant un seul-en-scène ? Rêver le temps d’un opéra ou d’un spectacle de danse, assister à une pièce de théâtre, ou vous laisser conter une histoire ?"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        id="category-description"
      >
        <h2
          className="nav-result-title"
        >
          Applaudir
        </h2>
        <span
          className="nav-result-description"
        >
          Voulez-vous suivre un géant de 12 mètres dans la ville ? Rire devant un seul-en-scène ? Rêver le temps d’un opéra ou d’un spectacle de danse, assister à une pièce de théâtre, ou vous laisser conter une histoire ?
        </span>
      </div>,
      "id": "nav-results-header",
      "style": Object {
        "backgroundImage": "url(http://localhost/icons/img-Applaudir-L.jpg)",
      },
      "title": "Liste des offres de type Applaudir",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="nav-result-title"
          >
            Applaudir
          </h2>,
          <span
            className="nav-result-description"
          >
            Voulez-vous suivre un géant de 12 mètres dans la ville ? Rire devant un seul-en-scène ? Rêver le temps d’un opéra ou d’un spectacle de danse, assister à une pièce de théâtre, ou vous laisser conter une histoire ?
          </span>,
        ],
        "id": "category-description",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Applaudir",
            "className": "nav-result-title",
          },
          "ref": null,
          "rendered": "Applaudir",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Voulez-vous suivre un géant de 12 mètres dans la ville ? Rire devant un seul-en-scène ? Rêver le temps d’un opéra ou d’un spectacle de danse, assister à une pièce de théâtre, ou vous laisser conter une histoire ?",
            "className": "nav-result-description",
          },
          "ref": null,
          "rendered": "Voulez-vous suivre un géant de 12 mètres dans la ville ? Rire devant un seul-en-scène ? Rêver le temps d’un opéra ou d’un spectacle de danse, assister à une pièce de théâtre, ou vous laisser conter une histoire ?",
          "type": "span",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          id="category-description"
        >
          <h2
            className="nav-result-title"
          >
            Applaudir
          </h2>
          <span
            className="nav-result-description"
          >
            Voulez-vous suivre un géant de 12 mètres dans la ville ? Rire devant un seul-en-scène ? Rêver le temps d’un opéra ou d’un spectacle de danse, assister à une pièce de théâtre, ou vous laisser conter une histoire ?
          </span>
        </div>,
        "id": "nav-results-header",
        "style": Object {
          "backgroundImage": "url(http://localhost/icons/img-Applaudir-L.jpg)",
        },
        "title": "Liste des offres de type Applaudir",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="nav-result-title"
            >
              Applaudir
            </h2>,
            <span
              className="nav-result-description"
            >
              Voulez-vous suivre un géant de 12 mètres dans la ville ? Rire devant un seul-en-scène ? Rêver le temps d’un opéra ou d’un spectacle de danse, assister à une pièce de théâtre, ou vous laisser conter une histoire ?
            </span>,
          ],
          "id": "category-description",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Applaudir",
              "className": "nav-result-title",
            },
            "ref": null,
            "rendered": "Applaudir",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Voulez-vous suivre un géant de 12 mètres dans la ville ? Rire devant un seul-en-scène ? Rêver le temps d’un opéra ou d’un spectacle de danse, assister à une pièce de théâtre, ou vous laisser conter une histoire ?",
              "className": "nav-result-description",
            },
            "ref": null,
            "rendered": "Voulez-vous suivre un géant de 12 mètres dans la ville ? Rire devant un seul-en-scène ? Rêver le temps d’un opéra ou d’un spectacle de danse, assister à une pièce de théâtre, ou vous laisser conter une histoire ?",
            "type": "span",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
